core/src/main/java/com/graphhopper/routing/util/EncodingManager.java

```diff
@@ -211,23 +205,19 @@ private void addEncoder(VehicleEncodedValues encoder) {
         encoder.setEncodedValueLookup(this);
         List<EncodedValue> list = new ArrayList<>();
         encoder.createEncodedValues(list);
-        for (EncodedValue ev : list)
-            addEncodedValue(ev, true);
+        for (EncodedValue ev : list) {
+            addEncodedValue(ev);
+        }
         list = new ArrayList<>();
         encoder.createTurnCostEncodedValues(list);
         for (EncodedValue ev : list)
             addTurnCostEncodedValue(ev);
         edgeEncoders.add(encoder);
     }
 
-    private void addEncodedValue(EncodedValue ev, boolean withNamespace) {
-        String normalizedKey = ev.getName().replaceAll(SPECIAL_SEPARATOR, "_");
```

Comment:
And it did not prevent us from adding the same EV name either...
For example when we added an EV with `ev.getName() = "car$access" twice (from different FlagEncoders that return the same name EV in createEncodedValues) the `normalizedKey` would be "car_access" (except it was not but that does not even matter here). But since we add the EV with the key `ev.getName()` the second time we add it `hasEncodedValue(normalizedKey)` would still be false, so there was nothing that stopped us from adding multiple encoded values with the same name, each time overwriting the previous one.